File: FitoGrano/CS/Functions/mdRegistrySetting.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using System.Linq;
using System.Drawing;
using System.Diagnostics;
using Microsoft.VisualBasic;
using System.Xml.Linq;
using System.Windows.Forms;
// End of VB project level imports

using DevExpress.XtraEditors;
using Microsoft.Win32;
using FitoGrano;

// класс для работы с реестром (для настроек приложения)
namespace FitoGrano
{
	internal sealed class mdRegistrySetting
	{
#region Работа с реестром
		// Чтение из реестра
		public static dynamic RegistryRead(string SectionName, string KeyName, string NothingValue = "", RegistryValueKind NothingDataType = RegistryValueKind.String)
			{
			// если раздела не существует, но нет ключа вернется значение по-умолчанию
			// Проверка существования раздела
			string iSec = System.Convert.ToString(Registry.CurrentUser.CreateSubKey(SectionName).GetValue(KeyName, null)); // Возвращаем значение из реестра
			if (string.IsNullOrEmpty(iSec)) // Если раздела и ключа не существует
			{
				// они будут созданы автоматически
				RegistryWrite(SectionName, KeyName, NothingValue, NothingDataType);
			}
			// если существуют, вернуться значения из реестра
			return Registry.CurrentUser.CreateSubKey(SectionName).GetValue(KeyName, NothingValue);
		}
		// Запись в реестр
		public static void RegistryWrite(string SectionName, string KeyName, string Value, RegistryValueKind DataType = RegistryValueKind.String)
			{
			// Если ни раздела, ни ключа не существует, он будет создан
			(new Microsoft.VisualBasic.Devices.Computer()).Registry.CurrentUser.CreateSubKey(SectionName, false).SetValue(KeyName, Value, DataType); // Записываем текст в раздел Test
		}
		// Создание раздела
		public static void RegistryCreateSection(string SectionName)
		{
			(new Microsoft.VisualBasic.Devices.Computer()).Registry.CurrentUser.CreateSubKey(SectionName, RegistryKeyPermissionCheck.ReadWriteSubTree); // Создаем новый раздел в реестре с доступом на запись
		}
		// Удаление раздела
		public static void RegistryDeleteSection(string SectionName)
		{
			try
			{
				(new Microsoft.VisualBasic.Devices.Computer()).Registry.CurrentUser.DeleteSubKeyTree(SectionName, true); // Удаляем раздел из реестра
			}
			catch (Exception e)
			{
				XtraMessageBox.Show(e.Message, Application.ProductName, MessageBoxButtons.OK, MessageBoxIcon.Error);
			}
		}
		// Удаление ключа из ветки реестра
		public static dynamic RegistryDeleteKey(string SectionName, string KeyName)
		{
			try
			{
				(new Microsoft.VisualBasic.Devices.Computer()).Registry.CurrentUser.OpenSubKey(SectionName, true).DeleteValue(KeyName, true);
			}
			catch (Exception eX)
			{
				XtraMessageBox.Show(eX.Message, Application.ProductName, MessageBoxButtons.OK, MessageBoxIcon.Error);
			}
		}
#endregion
	}
	
}

File: FitoGrano/Functions/mdRegistrySetting.vb
Imports DevExpress.XtraEditors
Imports Microsoft.Win32
' класс для работы с реестром (для настроек приложения)
Friend Module mdRegistrySetting
#Region "Работа с реестром"
    ' Чтение из реестра
    Public Function RegistryRead(ByVal SectionName As String,
                                 ByVal KeyName As String,
                                 Optional ByVal NothingValue As String = "",
                                 Optional ByVal NothingDataType As RegistryValueKind = RegistryValueKind.String)
        ' если раздела не существует, но нет ключа вернется значение по-умолчанию
        ' Проверка существования раздела
        Dim iSec As String = Registry.CurrentUser.CreateSubKey(SectionName).GetValue(KeyName, Nothing) ' Возвращаем значение из реестра
        If iSec = Nothing Then ' Если раздела и ключа не существует
            ' они будут созданы автоматически
            RegistryWrite(SectionName, KeyName, NothingValue, NothingDataType)
        End If
        ' если существуют, вернуться значения из реестра
        Return Registry.CurrentUser.CreateSubKey(SectionName).GetValue(KeyName, NothingValue)
    End Function
    ' Запись в реестр
    Public Sub RegistryWrite(ByVal SectionName As String,
                             ByVal KeyName As String,
                             ByVal Value As String,
                             Optional ByVal DataType As RegistryValueKind = RegistryValueKind.String)
        ' Если ни раздела, ни ключа не существует, он будет создан
        My.Computer.Registry.CurrentUser.CreateSubKey(SectionName, False).SetValue(KeyName, Value, DataType) ' Записываем текст в раздел Test
    End Sub
    ' Создание раздела
    Public Sub RegistryCreateSection(ByVal SectionName As String)
        My.Computer.Registry.CurrentUser.CreateSubKey(SectionName, RegistryKeyPermissionCheck.ReadWriteSubTree) ' Создаем новый раздел в реестре с доступом на запись
    End Sub
    ' Удаление раздела
    Public Sub RegistryDeleteSection(ByVal SectionName As String)
        Try
            My.Computer.Registry.CurrentUser.DeleteSubKeyTree(SectionName, True) ' Удаляем раздел из реестра
        Catch e As Exception
            XtraMessageBox.Show(e.Message, Application.ProductName, MessageBoxButtons.OK, MessageBoxIcon.Error)
        End Try
    End Sub
    ' Удаление ключа из ветки реестра
    Public Function RegistryDeleteKey(ByVal SectionName As String, ByVal KeyName As String)
        Try
            My.Computer.Registry.CurrentUser.OpenSubKey(SectionName, True).DeleteValue(KeyName, True)
        Catch eX As Exception
            XtraMessageBox.Show(eX.Message, Application.ProductName, MessageBoxButtons.OK, MessageBoxIcon.Error)
        End Try
    End Function
#End Region
End Module
